unstructured/ingest/connector/astra.py

```diff
@@ -123,3 +130,117 @@ def normalize_dict(self, element_dict: dict) -> dict:
             "content": element_dict.pop("text", None),
             "metadata": element_dict,
         }
+
+
+@dataclass
+class AstraIngestDoc(IngestDocCleanupMixin, BaseSingleIngestDoc):
+    connector_config: SimpleAstraConfig
+    metadata: t.Dict[str, str] = field(default_factory=dict)
+    registry_name: str = "astra"
+
+    @property
+    def filename(self):
+        return (
+            Path(self.processor_config.output_dir)
+            / f"{self.metadata['_id']}.txt"
+        ).resolve()
+ 
+    @property
+    def _output_filename(self):
+        return Path(self.processor_config.output_dir) / f"{self.filename}.json"
+    
+    def update_source_metadata(self, **kwargs):
+        if not self.metadata:
+            self.source_metadata = SourceMetadata(
+                exists=False,
+            )
+            return
+        self.source_metadata = SourceMetadata(
+            exists=True,
+        )
+    
+    @SourceConnectionError.wrap
+    @requires_dependencies(["astrapy"], extras="astra")
+    @BaseSingleIngestDoc.skip_if_file_exists
+    def get_file(self):
```

Comment:
Since this should go to the download folder, I think we usually do something like this. Maybe check out how the mongodb is doing this.
```
        self.filename.parent.mkdir(parents=True, exist_ok=True)
        with open(self.filename, "w") as f:
```